desktop/src/com/mygdx/game/client/screen/TechnologyScreen.java

```diff
@@ -0,0 +1,82 @@
+package com.mygdx.game.client.screen;
+
+import com.artemis.World;
+import com.badlogic.gdx.Gdx;
+import com.badlogic.gdx.InputMultiplexer;
+import com.badlogic.gdx.ScreenAdapter;
+import com.badlogic.gdx.scenes.scene2d.Stage;
+import com.badlogic.gdx.utils.viewport.Viewport;
+import com.mygdx.game.client.GdxGame;
+import com.mygdx.game.client.TextureRenderer;
+import com.mygdx.game.client.input.CameraMoverInputProcessor;
+import com.mygdx.game.client.input.GameScreenUiInputAdapter;
+import com.mygdx.game.client.input.TechnologyInputAdapter;
+import com.mygdx.game.core.util.CompositeUpdatable;
+import lombok.NonNull;
+import lombok.extern.java.Log;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+@Singleton
+@Log
+public class TechnologyScreen extends ScreenAdapter {
+
+  private final CompositeUpdatable compositeUpdatable = new CompositeUpdatable();
+
+  private final World world;
+
+  private final GdxGame game;
+  private final Viewport viewport;
+  private final TextureRenderer renderer;
+
+  private final Stage stage;
+
+  @Inject
+  public TechnologyScreen(
+          @NonNull GdxGame game,
+          @NonNull World world,
+          @NonNull @Named("orthographic") Viewport viewport,
+          @NonNull TextureRenderer renderer,
+          @NonNull Stage stage
+          ) {
+    this.game = game;
+    this.world = world;
+    this.viewport = viewport;
+    this.renderer = renderer;
+    this.stage = stage;
+  }
+
+  @Override
+  public void show() {
+    log.info("Technology tree shown");
+
+    setUpInput();
+  }
+
+  @Override
+  public void render(float delta) {
+    compositeUpdatable.update(delta);
+    world.setDelta(delta);
+    world.process();
+    viewport.getCamera().update();
+    renderer.technologyRender();
+//    stage.draw();
```

Comment:
No trochę drzewo technologii mogło być już być robione w formie normalnych elementów UI, a nie gołych tekstur.

Reply:
Nie wiem w sumie co będzie łatwiejsze. Wydaje mi się, że Aktory z Scene2d mogą np. rosnąć gdyby była inna rozdzielczość.

Reply:
Typowe drzewo-drzewo może być jednak trochę problemem, bo Scene2D operuje bardziej na tabelkach. Więc może może być, ale pewnie jakiś edge-case będzie.

Reply:
Obsługują w łatwiejszy sposób kliknięcie na pewno, potrafią mieć osobne tekstury na różne stany, hover, pressed, enabled, disabled czy coś takiego. To się przyda na pewno.

Reply:
Hmmm okej, spróbuję to zrobić za pomocą aktorów

Reply:
możesz zobaczyć czy da się podpasować do tego jakąś architekturę UI - MVP, MVC, MVVM - nie wiem co będzie najlepsze - możiwe, że MVVM jest overkillem jak nie ma cyklu życia androida i starczy MVP albo MVC, ale nie wiem na czym one do końca polegają 

jest też MVI, ale to znam tylko z nazwy